### DISCUSSION TITLE
How to read barcode from scan in PDF?

### DISCUSSION THREAD

**Discussion by drnicolas** [zxing-cpp/zxing-cpp#789] [General]
I am experimenting with zxing_cpp and trying to decode barcodes from a scanned PDF
The page is read with PyPDF2.PDFReader
but zxingcpp.read_Barcodes fails with PyPDF2.PageObject does not support the buffer protocol

**Comment by axxel** [comment]
The `zxing-cpp` library works on image data, i.e. a 2-D array of pixels. You need to render your PDF page into an image and feed that into `zxingcpp.read_barcodes`. Here is a list of 5 available python libraries that can help you with that: https://levelup.gitconnected.com/4-python-libraries-to-convert-pdf-to-images-7a09eba83a09
